server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -456,7 +468,9 @@ public short offsetForLeaderEpochRequestVersion() {
     }
 
     public short listOffsetRequestVersion() {
-        if (this.isAtLeast(IBP_3_5_IV0)) {
+        if (this.isAtLeast(IBP_3_9_IV0)) {
+            return 9;
```

Comment:
This is problematic. The replication thread uses this method to select the version for ListOffsetRequest, instead of using the supported version from ApiResponse. Since this version is unstable, the receiving broker will fail the request with the default configuration of `unstable.api.versions.enable`.

Reply:
IBP_3_9_IV0 cannot be set by regular users yet. It's not a stable MV. It's fine for unstable MVs to use unstable RPCs.

Reply:
Good point.